### PR TITLE
 Convert Credentials to Kotlin

### PR DESCRIPTION
**Pull request by Egorand**

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/Credentials.kt [hunk 1]
```diff
@@ -13,26 +13,22 @@
...
   /** Returns an auth credential for the Basic scheme. */
-  public static String basic(String username, String password) {
-    return basic(username, password, ISO_8859_1);
-  }
-
-  public static String basic(String username, String password, Charset charset) {
-    String usernameAndPassword = username + ":" + password;
-    String encoded = ByteString.encodeString(usernameAndPassword, charset).base64();
-    return "Basic " + encoded;
+  @JvmStatic @JvmOverloads fun basic(
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/Credentials.kt]: Nice, kotlin is a good win for this stuff.

**Comment by Egorand** [comment]
Build failure looks unrelated to the PR

**Comment by yschimke** [comment]
@Egorand Yep, testing with CircleCI and hitting two issues.  It doesn't checkout the hpack git submodule automatically, and it is timing out in 2 tests of MockWebServer.  The signal from Travis is best for the moment.

**Comment by Egorand** [comment]
Got it @yschimke! Merging as is based on Travis build results.

**Pull request merged**